src/main/java/org/example/calculator/model/Calculator.java

```diff
@@ -0,0 +1,47 @@
+package org.example.calculator.model;
+
+import java.util.Arrays;
+import java.util.function.Function;
+
+public class Calculator {
+
+    private static final String DEFAULT_DELIMITER = ",|:";
+    private static final String CUSTOM_DELIMITER_REGEX = "//(.)\n";
+    private static final String CUSTOM_OPERANDS_REGEX = "//.\n(.+)";
+    private final String string;
+
+    public Calculator(String string) {
+        this.string = string == null ? "" : string;
+    }
+
+    public int calculate() {
+        String delimiter = CalculatorStringUtils.extractDelimiterWithRegex(string, CUSTOM_DELIMITER_REGEX, DEFAULT_DELIMITER);
+        String operandString = CalculatorStringUtils.extractOperandStringWithRegex(string, CUSTOM_OPERANDS_REGEX);
+        String[] operands = splitOperandStringWithDelimiter(operandString, delimiter);
+
+        return getStringOperandsSum(operands);
+    }
+
+    private int getStringOperandsSum(String[] operands) {
+        return Arrays.stream(operands)
+                .map(emptyStringToZero())
+                .map(Integer::parseInt)
+                .peek(this::validateNonNegativeInteger)
+                .mapToInt(Integer::intValue)
+                .sum();
+    }
+
+    private Function<String, String> emptyStringToZero() {
+        return s -> s.isBlank() ? "0" : s;
+    }
+
+    private String[] splitOperandStringWithDelimiter(String operandString, String delimiter) {
+        return operandString.split(delimiter);
+    }
+
+    private void validateNonNegativeInteger(Integer integer) {
+        if (integer < 0) {
+            throw new IllegalArgumentException(String.format("negatives not allowed: %d", integer));
+        }
+    }
```

Comment:
메서드는 호출되는 순서대로 위치시키면 조금 더 코드를 파악하기가 수월할 것 같아요. 😃

```suggestion
    private String[] splitOperandStringWithDelimiter(String operandString, String delimiter) {
        return operandString.split(delimiter);
    }

    private int getStringOperandsSum(String[] operands) {
        return Arrays.stream(operands)
                .map(emptyStringToZero())
                .map(Integer::parseInt)
                .peek(this::validateNonNegativeInteger)
                .mapToInt(Integer::intValue)
                .sum();
    }

    private Function<String, String> emptyStringToZero() {
        return s -> s.isBlank() ? "0" : s;
    }

    private void validateNonNegativeInteger(Integer integer) {
        if (integer < 0) {
            throw new IllegalArgumentException(String.format("negatives not allowed: %d", integer));
        }
    }
```